aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -328,6 +334,27 @@ public SdkHttpFullRequest sign(
     return mutableRequest.build();
   }
 
+  /**
+   * Only add body for DeleteObjectsRequest. Refer to
+   * https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html#API_DeleteObjects_RequestSyntax
+   */
+  private String body(SdkHttpFullRequest request) {
+    if (isDeleteObjectsRequest(request) && request.contentStreamProvider().isPresent()) {
+      try (InputStream is = request.contentStreamProvider().get().newStream()) {
+        return IoUtils.toUtf8String(is);
+      } catch (IOException e) {
+        throw new UncheckedIOException("Failed to set body for S3 sign request", e);
+      }
+    }
+
+    return null;
+  }
+
+  private boolean isDeleteObjectsRequest(SdkHttpFullRequest request) {
+    return request.method() == SdkHttpMethod.POST
+        && request.rawQueryParameters().containsKey("delete");
```

Comment:
S3 supports a lot of requests for other operations.  Are there any other combinations of requests that might meet this criteria?  Just wondering because we may be exposing the client for other operations and I want to make sure this uniquely identifies the bulk delete operation.

Reply:
I glanced over a few other common request types and couldn't find anything that would match this, but would be good to do a thorough check

Reply:
I wasn't able to find anything either (referencing https://docs.aws.amazon.com/AmazonS3/latest/API/API_Operations_Amazon_Simple_Storage_Service.html) , I'll do a double pass and also see if there's even more specific criteria for bulk deletes beyond the method type and query parameter.

Reply:
I did a double pass to make sure there wasn't any other request that could possibly match this criteria. 
As far as getting even more unique, we could actually perform an extraction of the DeleteObjectsRequest contents from the body (because we know no other request at the moment would send that). But this is probably too complex for what it brings considering what we have now looks to be unique already (we would need to add the jackson XML dependency to the signer client, etc).

Lmk if you have any concerns @danielcweeks @nastra